src/transformers/utils/bitsandbytes_utils.py

```diff
@@ -0,0 +1,158 @@
+    function is adapted from `set_module_tensor_to_device` function from accelerate that is adapted to support the
+    class `Int8Params` from `bitsandbytes`.
+
+    Args:
+        module (`torch.nn.Module`):
+            The module in which the tensor we want to move lives.
+        tensor_name (`str`): The full
+            name of the parameter/buffer.
+        device (`int`, `str` or `torch.device`):
+            The device on which to set the tensor.
+        value (`torch.Tensor`, *optional*):
+            The value of the tensor (useful when going from the meta device to any other device).
+    """
+    # Recurse if needed
+    if "." in tensor_name:
+        splits = tensor_name.split(".")
+        for split in splits[:-1]:
+            new_module = getattr(module, split)
+            if new_module is None:
+                raise ValueError(f"{module} has no attribute {split}.")
+            module = new_module
+        tensor_name = splits[-1]
+
+    if tensor_name not in module._parameters and tensor_name not in module._buffers:
+        raise ValueError(f"{module} does not have a parameter or a buffer named {tensor_name}.")
+    is_buffer = tensor_name in module._buffers
+    old_value = getattr(module, tensor_name)
+
+    if old_value.device == torch.device("meta") and device not in ["meta", torch.device("meta")] and value is None:
+        raise ValueError(f"{tensor_name} is on the meta device, we need a `value` to put in on {device}.")
+
+    if is_buffer:
+        has_fp16_weights = None
+    else:
+        has_fp16_weights = getattr(module._parameters[tensor_name], "has_fp16_weights", None)
+
+    if has_fp16_weights is not None:
+        param = module._parameters[tensor_name]
+        if param.device.type != "cuda":
+            with torch.no_grad():
+                if value is None:
+                    new_value = old_value.to(device)
+                elif isinstance(value, torch.Tensor):
+                    new_value = value.to("cpu")
+                else:
+                    new_value = torch.tensor(value, device="cpu")
+                new_value = bnb.nn.Int8Params(new_value, requires_grad=False, has_fp16_weights=has_fp16_weights).to(
+                    device
+                )
+                module._parameters[tensor_name] = new_value
+    else:
+        with torch.no_grad():
+            if value is None:
+                new_value = old_value.to(device)
+            elif isinstance(value, torch.Tensor):
+                new_value = value.to(device)
+            else:
+                new_value = torch.tensor(value, device=device)
+        if is_buffer:
+            module._buffers[tensor_name] = new_value
+        else:
+            new_value = nn.Parameter(new_value, requires_grad=old_value.requires_grad)
+            module._parameters[tensor_name] = new_value
+
+
+@contextmanager
+def init_empty_weights_8bit(include_buffers=False):
```

Comment:
As said above, we shouldn't need to change the decorator for `init_empty_weights` as the linear layers have not been replaced yet, so there is no parameter that will have `has_fp16_weights` attributes.

Reply:
I think that you are right, let me double check, we might not need this function at all

Reply:
I can confirm that you are right, proposed a modification in b92c25c3835f37667ee033a2b406846da06f16f4